### ISSUE TITLE
Refactor Property class

### ISSUE THREAD

**Issue by ra3xdh** [ra3xdh/qucs_s#974]
The `Property`  (see `element.h`) class is inherited mostly unchanged from Qt3 application version and may benefit from the refactoring. I propose the following things:

* Add a field containing property type: file, equation or usual property
* Add a field containing allowed simulator type: SPICE, Qucsator, etc.

The implementation of the second will allow to hide simulator-incompatible device properties from user to prevent its accidental setting and usage.

**Comment by ra3xdh** [comment]
Transferred to the next release cycle.

**Comment by wawuwo** [comment]
>Add a field containing allowed simulator type: SPICE, Qucsator, etc.

@ra3xdh are the members of `spicecompat::Simulator` meant here?

**Comment by wawuwo** [comment]
@ra3xdh could please take a look at previous comment?

**Comment by ra3xdh** [comment]
> are the members of spicecompat::Simulator meant here?

Yes, it this enum. The logic should be following. If the property is not simulator compatible hide it from component properties dialog and schematic view. Otherwise it may confuse user. The default value should `simAll`

**Comment by ra3xdh** [comment]
> Add a field containing property type: file, equation or usual property

The idea behind this is to get rid of hack with "Symbol" property name like in `param_sweep.cpp` line 68 and other places.
